File: 2025-09-25 - ListaPrioridadePQ/Exemplo_C#/Node.cs
// ------------------------------------------------------------------------
// Node.cs
// Classe que representa um nó da lista de prioridade
// Todos os métodos e propriedades em camelCase
// ------------------------------------------------------------------------
public class Node
{
    // Campos privados
    private string codigoDisciplina;
    private string nomeDisciplina;
    private int cargaHoraria;
    private int semestre;

    // Construtor
    public Node(string codigoDisciplina, string nomeDisciplina, int cargaHoraria, int semestre)
    {
        this.codigoDisciplina = codigoDisciplina;
        this.nomeDisciplina = nomeDisciplina;
        this.cargaHoraria = cargaHoraria;
        this.semestre = semestre;
    }

    // Propriedades públicas (camelCase)
    public string codigoDisciplinaProp
    {
        get { return codigoDisciplina; }
        set { codigoDisciplina = value; }
    }

    public string nomeDisciplinaProp
    {
        get { return nomeDisciplina; }
        set { nomeDisciplina = value; }
    }

    public int cargaHorariaProp
    {
        get { return cargaHoraria; }
        set { cargaHoraria = value; }
    }

    public int semestreProp
    {
        get { return semestre; }
        set { semestre = value; }
    }

    // Método toStringNode em camelCase
    public string toStringNode()
    {
        return $"{codigoDisciplina} – {nomeDisciplina} ({cargaHoraria}h) - Semestre {semestre}";
    }
}

File: 2025-09-25 - ListaPrioridadePQ/Exemplo_Python/Node.py
# ------------------------------------------------------------------------
# node.py
# Classe que representa um nó da lista de prioridade
# Todos os métodos e atributos em camelCase
# ------------------------------------------------------------------------
class Node:
    def __init__(self, codigoDisciplina, nomeDisciplina, cargaHoraria, semestre):
        self._codigoDisciplina = codigoDisciplina
        self._nomeDisciplina = nomeDisciplina
        self._cargaHoraria = cargaHoraria
        self._semestre = semestre

    # ---------------------------------------------------------------------
    # Getters e setters em camelCase
    # ---------------------------------------------------------------------
    @property
    def codigoDisciplinaProp(self):
        return self._codigoDisciplina

    @codigoDisciplinaProp.setter
    def codigoDisciplinaProp(self, value):
        self._codigoDisciplina = value

    @property
    def nomeDisciplinaProp(self):
        return self._nomeDisciplina

    @nomeDisciplinaProp.setter
    def nomeDisciplinaProp(self, value):
        self._nomeDisciplina = value

    @property
    def cargaHorariaProp(self):
        return self._cargaHoraria

    @cargaHorariaProp.setter
    def cargaHorariaProp(self, value):
        self._cargaHoraria = value

    @property
    def semestreProp(self):
        return self._semestre

    @semestreProp.setter
    def semestreProp(self, value):
        self._semestre = value

    # ---------------------------------------------------------------------
    # Método toStringNode em camelCase
    # ---------------------------------------------------------------------
    def toStringNode(self):
        return f"{self._codigoDisciplina} – {self._nomeDisciplina} ({self._cargaHoraria}h) - Semestre {self._semestre}"

    # Para PriorityQueue, precisamos que os nós sejam comparáveis pelo semestre
    def __lt__(self, other):
        return self._semestre < other._semestre
